src/gjk/power_to_energy.py

```diff
@@ -0,0 +1,173 @@
+        last_power_before_current_hour = power_before_current_hour[-1]
+    else:
+        last_power_before_current_hour = 0
+
+    for hour in range(num_hours):
+        
+        current_hour = first_hour.add(hours=hour)
+        next_hour = first_hour.add(hours=hour+1)
+
+        # Isolate data for the given hour
+        times_for_this_hour = [time_data[f'{p}'][i] for i in range(len(time_data[f'{p}'])) if time_data_hours[i]==current_hour]
+        powers_for_this_hour = [power_data[f'{p}'][i] for i in range(len(power_data[f'{p}'])) if time_data_hours[i]==current_hour]
+
+        # Add points at the start and end of the hour
+        times_for_this_hour = [current_hour.timestamp()*1000] + times_for_this_hour + [next_hour.timestamp()*1000]
+        if powers_for_this_hour:
+            powers_for_this_hour = [last_power_before_current_hour] + powers_for_this_hour + [powers_for_this_hour[-1]]
+            last_power_before_current_hour = powers_for_this_hour[-1] # for the next hour
+        else:
+            powers_for_this_hour = [last_power_before_current_hour] + powers_for_this_hour + [last_power_before_current_hour]
+
+        # Compute the energy for the given hour
+        time_diff_hours = [(y-x)/1000/60/60 for x,y in zip(times_for_this_hour[:-1], times_for_this_hour[1:])]
+        energy = round(sum([powers_for_this_hour[i] * time_diff_hours[i] for i in range(len(time_diff_hours))]),2)
+        energy_results[f'{p}'].append(0 if energy<0 else energy)
+
+# ------------------------------------------
+# Convert the results to a dataframe
+# ------------------------------------------
+
+df = pd.DataFrame({'id':[], 'hour_start_s':[], 'channel_name':[], 'watt_hours':[], 'g_node_alias':[]})
+for p in power_channels:
+    for h in range(len(hour_data)):
+
+        g_node = 'd1.isone.ver.keene.beech'
+        channel = p+'-energy'
+        hour_start = int(hour_data[h].timestamp())
+        value = int(energy_results[f'{p}'][h])
+        id = f"{g_node}_{hour_start}_{channel}"
+
+        row = [id, hour_start, channel, value, g_node]
+        if value>0 or record_zeros:
+            df.loc[len(df)] = row
+
+df.sort_values(by='id', inplace=True)
+df.reset_index(drop=True, inplace=True)
+print(df)
+
+# When running on two different computers
+# df.to_csv('thomas.csv', index=False)
+# df = pd.read_csv('/Users/thomas/Desktop/thomas.csv')
+
+# ------------------------------------------
+# Add the table to the database
+# ------------------------------------------
+
+def add_row(row_data):
+
+    energy_row_sqlalchemy = GWPowerSQLAlchemy(
+        id = row_data[0],
+        hour_start_s = row_data[1], 
+        channel_name = row_data[2],
+        watt_hours = int(row_data[3]), # remove this later
+        g_node_alias = row_data[4])
+    Session = sessionmaker(bind=engine)
+    session = Session()
+    session.add(energy_row_sqlalchemy)
+    session.commit()
+    session.close()
+
+Base = declarative_base()
+class GWPowerSQLAlchemy(Base):
```

Comment:
1.  Move this into models/ following the pattern of the other tables. 
2. from gjk.models.message import Base 
3. Name should not include power, as this is energy
4. Follow the naming conventions. Eg ReadingSql, DataChannelSql ... so call this one HourlyEnergySql